pyro/poutine/trace_struct.py

```diff
@@ -145,17 +148,24 @@ def compute_log_prob(self, site_filter=lambda name, site: True):
         """
         for name, site in self.nodes.items():
             if site["type"] == "sample" and site_filter(name, site):
-                try:
+                if "log_prob" in site:
```

Comment:
We don't need the `if` block and can just do `if "log_prob" not in site:`. Do you still need the reraising below with this change - does this not give the full traceback?

Reply:
The reason I am reraising is to add the name of the site to the message from the validation error. I will change the block to `if not` as suggested. 😄

Reply:
I see, but the traceback should also specify for which particular distribution this fails. Is that not the case?

e.g. I get something like below, which shows the problem is with the beta's log_prob. 

```
Traceback (most recent call last):
  File "/Users/npradhan/workspace/pyro_dev/pyro/examples/ex1.py", line 14, in <module>
    print(tr.compute_log_prob())
  File "/Users/npradhan/workspace/pyro_dev/pyro/pyro/poutine/trace_struct.py", line 149, in compute_log_prob
    log_p = site["fn"].log_prob(site["value"], *site["args"], **site["kwargs"])
  File "/Users/npradhan/miniconda2/envs/pytorch-36/lib/python3.6/site-packages/torch/distributions/beta.py", line 58, in log_prob
    self._validate_sample(value)
  File "/Users/npradhan/miniconda2/envs/pytorch-36/lib/python3.6/site-packages/torch/distributions/distribution.py", line 221, in _validate_sample
    raise ValueError('The value argument must be within the support')
ValueError: The value argument must be within the support
```

Reply:
The last few lines in the exception trace in my modified code shows:

```
File ".../pyro-dev-env-py2/lib/python2.7/site-packages/torch/distributions/categorical.py", line 94, in log_prob
    self._validate_sample(value)
  File ".../pyro-dev-env-py2/lib/python2.7/site-packages/torch/distributions/distribution.py", line 221, in _validate_sample
    raise ValueError('The value argument must be within the support')
ValueError: Error while computing log_prob at site 'obs': The value argument must be within the support
```
Which points to the `categorical` file as cause. I guess I could also include the distribution's name in the error message if needed. I will see how that can be done.

Reply:
Apologies for any confusion! My suggestion was to only remove the `try...catch KeyError` block (that you seem to have done), and not bother with catching and re-raising the ValueError, since I believe that the traceback will contain the necessary information to aid debugging. 

I believe that what you are looking to do is prettify the error message reported so that it is clear what is happening without having to go through the traceback. That seems fine to me!

Reply:
Ah, no worries! I usually meet the issues when I use the same distribution multiple times, and I have to figure out which site actually is causing the error 😄

Reply:
Thanks for explaining, makes sense.